src/lib/eslint/ESLintEngine.ts

```diff
@@ -21,74 +109,137 @@ export class ESLintEngine implements RuleEngine {
+			}
+			category.paths.push(docs.url);
+
+			const r = {
+				engine: ESLintEngine.NAME,
+				sourcepackage: ESLintEngine.NAME,
+				name: key,
+				description: docs.description,
+				categories: [docs.category],
+				rulesets: [docs.category],
+				languages: [ESLintEngine.JAVASCRIPT_LANGUAGE],
+				defaultEnabled: docs.recommended,
+				url: docs.url
+			};
+			if (key.startsWith("@typescript")) {
+				// Typescript is superset of javascript, so add it on top here.
+				r.languages.push(ESLintEngine.TYPESCRIPT_LANGUAGE);
+			}
+			rules.push(r)
+		});
+
+		catalog.categories = Array.from(categoryMap.values());
+		catalog.rules = rules;
+		return Promise.resolve(catalog);
 	}
 
 	public async getAll(): Promise<Rule[]> {
-		// PmdCatalogWrapper is a layer of abstraction between the commands and PMD, facilitating code reuse and other goodness.
-		this.logger.trace('Getting PMD rules.');
+		this.logger.trace('Getting eslint rules.');
 		return Promise.resolve([]);
 	}
 
-	public async run(paths: PathGroup[], target: string[]): Promise<string> {
-		this.logger.trace(`About to run eslint rules. Target count: ${target.length}, filter count: ${paths.length}`);
-		try {
-			/*
-			const linter = new Linter();
-
-			const sample =
-				{
-					fatal: false,
-					ruleId: "semi",
-					severity: 2,
-					line: 1,
-					column: 23,
-					message: "Expected a semicolon.",
-					fix: {
-						range: [1, 15],
-						text: ";"
-					}
-				};
-
-			const messages = linter.verify("var foo;", {
-				rules: {
-					semi: 2
-				}
-			}, {filename: "foo.js"});
-*/
+	public async run(ruleGroups: RuleGroup[], rules: Rule[], targets: string[]): Promise<RuleResult[]> {
+		const targetPaths: string[] = targets;
+		// If we didn't find any paths, we're done.
+		if (targetPaths == null || targetPaths.length === 0) {
+			this.logger.trace('No matching target files found. Nothing to execute.');
+			return [];
+		}
 
-			const paths: string[] = [];
-			// If we didn't find any paths, we're done.
-			if (paths == null || paths.length === 0) {
-				this.logger.trace('No Rule paths found. Nothing to execute.');
-				return '';
-			}
-			const [violationsFound, stdout] = [null, null];
-
-			if (violationsFound) {
-				this.logger.trace('Found rule violations.');
-				// If we found any violations, they'll be in an XML document somewhere in stdout, which we'll need to find and process.
-				const xmlStart = stdout.indexOf('<?xml');
-				const xmlEnd = stdout.lastIndexOf('</pmd>') + 6;
-				const ruleViolationsXml = stdout.slice(xmlStart, xmlEnd);
-
-				this.logger.trace(`Rule violations in the original XML format: ${ruleViolationsXml}`);
-				return ruleViolationsXml;
-			} else {
-				// If we didn't find any violations, we can just return an empty string.
-				this.logger.trace('No rule violations found.');
-				return '';
-			}
+		if (rules.length === 0) {
+			this.logger.trace('No matching eslint rules found. Nothing to execute.');
```

Comment:
Another message to surface to users.

Reply:
Same comment as above.  This is a valid use case for the engine not to have any matching targets.  It is not an error case.  The trace is there for debugging if something goes wrong.